### PR TITLE
Import ovsd from https://github.com/berlin-open-wireless-lab/package-ovsd

### PR DESCRIPTION
**Pull request by dangowrt**
Add Open vSwtich configuration daemon for netifd.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>

### CODE REVIEW

File: net/ovsd/files/etc/init.d/ovsd [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+START=19
+USE_PROCD=1
+
+start_service() {
+        procd_open_instance
+        procd_set_param command /sbin/ovsd
+        procd_set_param respawn
+	procd_set_param watch ovs
+
```
Comment (aparcar) [inline net/ovsd/files/etc/init.d/ovsd:11]: this could be beautified

File: net/ovsd/Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://gitlab.hhi.fraunhofer.de/wn-ina/ovsd.git
+PKG_SOURCE_VERSION:=e6a07f27dd8a5893eb6110c69922077995b9b26a
+PKG_MIRROR_HASH:=9d903b3fecd8a3e102e1477efe479fc3851b1814ea0852e1dae39065b88cd157
+
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION)-$(PKG_SOURCE_VERSION).tar.gz
+
+PKG_MAINTAINER:=Arne Kappen <akappen@inet.tu-berlin.de>
```
Comment (aparcar) [inline net/ovsd/Makefile:15]: Ack from the maintainer?

File: net/ovsd/Makefile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+PKG_LICENSE:=GPL-2.0
```
Comment (BKPepe) [inline net/ovsd/Makefile:16]: 1. Old SPDX License Identifier.
Take a look here: https://spdx.org/licenses/

2. Add `PKG_LICENSE_FILES:=LICENSE`

File: net/ovsd/Makefile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+include $(INCLUDE_DIR)/cmake.mk
+
+
+define Package/ovsd
+  SECTION:=net
+  SUBMENU:=Open vSwitch
+  CATEGORY:=Network
+  DEPENDS:=+openvswitch +libubox +libubus +netifd
+  TITLE:=Open vSwtich configuration daemon for netifd
+  MAINTAINER:=Arne Kappen <akappen@inet.tu-berlin.de>
```
Comment (BKPepe) [inline net/ovsd/Makefile:30]:
```suggestion
```

This is already here https://github.com/openwrt/packages/pull/15316/files#diff-b0352764a5c4bbbe3fad4fb194cd5becf1e50649c83872ee0f46c5d39c9dd15aR15

File: net/ovsd/Makefile [hunk 5]
```diff
@@ -0,0 +1,43 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=ovsd
+PKG_VERSION:=2020-10-12
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://gitlab.hhi.fraunhofer.de/wn-ina/ovsd.git
+PKG_SOURCE_VERSION:=e6a07f27dd8a5893eb6110c69922077995b9b26a
```
Comment (BKPepe) [inline net/ovsd/Makefile:9]: Why are you using old version? Can not we use the latest one?
In the suggestion, if you will approve it, it is updated to the latest version.

File: net/ovsd/Makefile [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+PKG_MIRROR_HASH:=9d903b3fecd8a3e102e1477efe479fc3851b1814ea0852e1dae39065b88cd157
+
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION)-$(PKG_SOURCE_VERSION).tar.gz
```
Comment (BKPepe) [inline net/ovsd/Makefile:13]:
```suggestion
PKG_SOURCE_PROTO:=git
PKG_SOURCE_URL:=https://gitlab.hhi.fraunhofer.de/wn-ina/ovsd
PKG_SOURCE_DATE:=2020-12-08
PKG_SOURCE_VERSION:=3a03476e2fececed3752ba4e68c92e71160c7db1
PKG_MIRROR_HASH:=7c6895b8aef8b3997ec86c2225792a443be9870985de7b7c01397cd97a6c7a25
```

File: net/ovsd/Makefile [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+PKG_MAINTAINER:=Arne Kappen <akappen@inet.tu-berlin.de>
+PKG_LICENSE:=GPL-2.0
+
+PKG_BUILD_PARALLEL:=1
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+
```
Comment (BKPepe) [inline net/ovsd/Makefile:23]:
```suggestion
```

File: net/ovsd/Makefile [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+define Package/ovsd
+  SECTION:=net
+  SUBMENU:=Open vSwitch
+  CATEGORY:=Network
+  DEPENDS:=+openvswitch +libubox +libubus +netifd
+  TITLE:=Open vSwtich configuration daemon for netifd
```
Comment (BKPepe) [inline net/ovsd/Makefile:29]:
```suggestion
  TITLE:=Open vSwtich configuration daemon for netifd
  URL:=https://gitlab.hhi.fraunhofer.de/wn-ina/ovsd
  DEPENDS:=+openvswitch +libubox +libubus +netifd
```

**Review by BKPepe — COMMENTED** [review]
I didn't do compile test, right now, but I noticed a few things, which could be improved. But you should get approval from Arne, that he is aware that you are adding him as @aparcar said.

File: net/ovsd/Makefile (hunk 2 above)
Comment (dangowrt) [reply net/ovsd/Makefile:15]: Yes, he was waiting for the needed changes in netifd to be merged which was only feasible after 21.02 branched off.

**Comment by BKPepe** [comment]
Closing as this is already in master branch.

**Pull request closed without merging**